File: src/jvm/compat-scala/src/test/scala/StrictlandSmokeTest.scala
import io.eventdriven.strictland.{MessageContract, MessageSnapshot, Json}
import org.scalatest.funsuite.AnyFunSuite

class StrictlandSmokeTest extends AnyFunSuite {

  test("library is consumable from Scala") {
    val contract = MessageContract.specification(Json.Jackson.defaults())
    val step = contract.`given`(MessageSnapshot.ofTypeNamed("SomeEvent"))

    assert(step != null)
  }
}

File: src/jvm/compat-kotlin/src/test/kotlin/StrictlandSmokeTest.kt
import io.eventdriven.strictland.Json
import io.eventdriven.strictland.MessageContract
import io.eventdriven.strictland.MessageSnapshot
import kotlin.test.Test
import kotlin.test.assertNotNull

class StrictlandSmokeTest {

    @Test
    fun `library is consumable from Kotlin`() {
        val contract = MessageContract.specification(Json.Jackson.defaults())
        val step = contract.given(MessageSnapshot.forMessageType("SomeEvent"))

        assertNotNull(step)
    }
}
